### DISCUSSION TITLE
How can I can I make a custom dynamic space drawer?

### DISCUSSION THREAD

**Discussion by DrEntropy** [mesa/mesa#2208] [Q&A] [answered]
It seems dependencies are not passed into a custom space drawer:
` space_drawer(model, agent_portrayal)`  is called in solara_viz.py line 73 without dependencies so my custom function is only called once at the beginning.  ( I am trying to draw a custom network layout).      How can I work around this to make it dynamic?  I have seen some posts on here referring to a version of space drawer that takes a `viz` as the first argument but I don't see this in the source code.  (main branch, at this date this is c96ee45) .

**Comment by rht** [comment, marked as answer]
Thanks for reporting. Fixed in #2209.

**Reply by DrEntropy** [reply]
Thanks!  I assumed I was just missing something, otherwise I would have opened an issue instead.  Cheers!

**Comment by EwoutH** [comment]
This is now [available](https://github.com/projectmesa/mesa/releases/tag/v3.0.0a2) in the latest Mesa 3.0 pre-release! Install with `pip install mesa --upgrade --pre`.

Thanks for reporting it.
